contracts/interchain-token-service/src/contract.rs

```diff
@@ -526,8 +510,57 @@ impl InterchainTokenService {
                     );
                 }
             }
-            Message::DeployInterchainToken(_) => {
-                todo!()
+            Message::DeployInterchainToken(DeployInterchainToken {
+                token_id,
+                name,
+                symbol,
+                decimals,
+                minter,
+            }) => {
+                ensure!(
+                    Self::token_id_config(env, token_id.clone()).is_err(),
+                    ContractError::TokenAlreadyDeployed
+                );
+
+                let token_metadata = TokenMetadata {
+                    name,
+                    symbol,
+                    decimal: decimals as u32,
+                };
+
+                ensure!(
+                    validate_token_metadata(token_metadata.clone()).is_ok(),
+                    ContractError::InvalidTokenMetaData
+                );
+
+                let minter_address = if let Some(minter) = minter {
+                    let addr = Address::from_xdr(env, &minter).expect("address conversion failed");
```

Comment:
This can fail due to a user input, should raise an error instead. Panics should only be used for unexpected scenarios, i.e due to coding errors (e.g. invalid initializations)

Reply:
I had this as `map_err` before to return a `ContractError`, but I discovered that when a byte string which does not represent a valid Soroban address is inputted, instead of returning a `ConversionError` as I expected, the Host simply panics with `Error(Value, InvalidInput)`. I'm not sure how to get this to return a meaningful error. Please advise.